### PR TITLE
Use default spotify color fetching

### PR DESCRIPTION
**Pull request by CharlieS1103**

### CODE REVIEW

**Comment by CharlieS1103** [comment]
I have a feeling this is going to bring our package size down substantially as well.

File: src/logic/Utils.ts [hunk 1]
```diff
@@ -307,43 +306,34 @@ export const initColorShiftLoop = (schemes: SchemeIni) => {
...
   return colorArray;
 };
 
-async function waitForAlbumArt(): Promise<HTMLImageElement | null> {
+async function waitForAlbumArt(): Promise<string | undefined> {
   // Only return when the album art is loaded
   return new Promise((resolve) => {
     setInterval(() => {
       const albumArt: HTMLImageElement | null = document.querySelector(".main-image-image.cover-art-image");
+      const albumArtSrc: string | undefined = albumArt?.src;
```
Comment (kyrie25) [inline src/logic/Utils.ts:334]: Why not use internal functions for this? Especially now you'll only need the URL.
```suggestion
      const albumArt = Spicetify.Player.data?.track?.metadata?.image_xlarge_url;
```

File: src/logic/Utils.ts [hunk 2]
```diff
@@ -307,43 +306,34 @@ export const initColorShiftLoop = (schemes: SchemeIni) => {
...
-
-  return "null";
+export const getColorFromImage = async (image: string) => {
+  const color = await (await Spicetify.colorExtractor(image)).VIBRANT_NON_ALARMING;
+  return color.substring(1);
 };
 
 export const generateColorPalette = async (mainColor: string, numColors: number) => {
   // Generate a palette from https://www.thecolorapi.com/id?hex=0047AB&rgb=0,71,171&hsl=215,100%,34%&cmyk=100,58,0,33&format=html
+  console.log(mainColor);
```
Comment (kyrie25) [inline src/logic/Utils.ts:316]: Cleanup.

File: src/logic/Utils.ts [hunk 3]
```diff
@@ -307,43 +306,34 @@ export const initColorShiftLoop = (schemes: SchemeIni) => {
...
   const mode = getLocalStorageDataFromKey(LOCALSTORAGE_KEYS.albumArtBasedColorMode);
   // Add a hyphen before any uppercase characters
   const modeStr = mode.replace(/([A-Z])/g, "-$1").toLowerCase();
   //fetch `https://www.thecolorapi.com/scheme?hex=${mainColor}&mode=${modeStr}&count=${numColors}`
   const palette = await  fetch(`https://www.thecolorapi.com/scheme?hex=${mainColor}&mode=${modeStr}&count=${numColors}`)
     .then((response) => response.json());
   // create an array of the hex values for the colors while also removing the #
+  console.log(palette);
```
Comment (kyrie25) [inline src/logic/Utils.ts:324]: Cleanup.

File: src/logic/Utils.ts [hunk 4]
```diff
@@ -354,16 +344,17 @@ export const initAlbumArtBasedColor = (scheme: ColourScheme) => {
   // and update the color scheme accordingly
   Spicetify.Player.addEventListener("songchange", async () => {
     await sleep(1000);
-    let albumArt: HTMLImageElement | null = document.querySelector(".main-image-image.cover-art-image");
+    const albumArt: HTMLImageElement | null = document.querySelector(".main-image-image.cover-art-image");
+    let albumArtSrc: string | undefined = albumArt?.src;
```
Comment (kyrie25) [inline src/logic/Utils.ts:348]: Ditto. Looks much cleaner and faster also.
```suggestion
    const albumArt = Spicetify.Player.data?.track?.metadata?.image_xlarge_url;
```

File: src/logic/Utils.ts [hunk 5]
```diff
@@ -354,16 +344,17 @@ export const initAlbumArtBasedColor = (scheme: ColourScheme) => {
...
-      albumArt = await waitForAlbumArt();
+    if (albumArtSrc == null || albumArt == null ||!albumArt.complete) {
+      albumArtSrc = await waitForAlbumArt();
     }
 
-    if (albumArt) {
+    if (albumArtSrc) {
       const numColors = new Set(Object.values(scheme)).size;
-      const mainColor = await getColorFromImage(albumArt, numColors);
+      const mainColor : string = await getColorFromImage(albumArtSrc);
```
Comment (kyrie25) [inline src/logic/Utils.ts:357]:
```suggestion
      const mainColor: string = await getColorFromImage(albumArtSrc);
```

**Review by kyrie25 — CHANGES_REQUESTED** [review]

**Comment by Tetrax-10** [comment]
can you expose chroma function to window object? so other themes and extensions dont need import them again
image

File: src/logic/Utils.ts [hunk 6]
```diff
@@ -307,43 +306,33 @@ export const initColorShiftLoop = (schemes: SchemeIni) => {
...
-  });
-
-  if (swatches.Vibrant) {
-    // remove the # from the hex
-    return swatches.Vibrant.hex.substring(1);
-  }
-
-  return "null";
+export const getColorFromImage = async (image: string) => {
+  const color = await (await Spicetify.colorExtractor(image)).VIBRANT_NON_ALARMING;
```
Comment (theRealPadster) [inline src/logic/Utils.ts:310]: Also don't need two awaits

**Comment by CharlieS1103** [comment]
Note do not merge till we review #404

**Comment by CharlieS1103** [comment]
Is this one good as well?

**Comment by CharlieS1103** [comment]
Screen Shot 2022-12-21 at 5 11 34 PM
@theRealPadster I don't have this issue

**Comment by theRealPadster** [comment]
> @theRealPadster I don't have this issue
Yeah my css changes fixed it 👍

**Review by theRealPadster — APPROVED** [review]
Probably good now?

**Pull request merged**